datasets/mt_eng_vietnamese/mt_eng_vietnamese.py

```diff
@@ -0,0 +1,127 @@
+            description=description,
+            version=datasets.Version("1.0.0"),
+            **kwargs,
+        )
+        self.language_pair = language_pair
+
+
+class MT_Eng_Vietnamese(datasets.GeneratorBasedBuilder):
+    """English Vietnamese machine translation dataset from IWSLT2015."""
+
+    BUILDER_CONFIGS = [
+        MT_Eng_ViConfig(
+            name="iwslt2015-vi-en",
+            language_pair=("vi", "en"),
+        ),
+        MT_Eng_ViConfig(
+            name="iwslt2015-en-vi",
+            language_pair=("en", "vi"),
+        ),
+    ]
+
+    def _info(self):
+        source, target = self.config.language_pair
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=datasets.Features(
+                {"translation": datasets.features.Translation(languages=self.config.language_pair)}
+            ),
+            supervised_keys=(source, target),
+            homepage="https://nlp.stanford.edu/projects/nmt/data/iwslt15.en-vi/",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        source, target = self.config.language_pair
+
+        files = {}
+        for split in ("train", "dev", "test"):
+            if split == "dev":
+                dl_dir_src = dl_manager.download_and_extract(_DATA_URL.format("tst2012", source))
+                dl_dir_tar = dl_manager.download_and_extract(_DATA_URL.format("tst2012", target))
+            if split == "dev":
+                dl_dir_src = dl_manager.download_and_extract(_DATA_URL.format("tst2013", source))
+                dl_dir_tar = dl_manager.download_and_extract(_DATA_URL.format("tst2013", target))
+            if split == "train":
+                dl_dir_src = dl_manager.download_and_extract(_DATA_URL.format(split, source))
+                dl_dir_tar = dl_manager.download_and_extract(_DATA_URL.format(split, target))
+
+            files[split] = {"source_file": dl_dir_src, "target_file": dl_dir_tar}
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TRAIN, gen_kwargs=files["train"]),
+            datasets.SplitGenerator(name=datasets.Split.VALIDATION, gen_kwargs=files["dev"]),
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs=files["test"]),
+        ]
+
+    def _generate_examples(self, source_file, target_file):
+        """This function returns the examples in the raw (text) form."""
+        with open(source_file, encoding="utf-8") as f:
+            source_sentences = f.read().split("\n")
+        with open(target_file, encoding="utf-8") as f:
+            target_sentences = f.read().split("\n")
+
+        source, target = self.config.language_pair
+        for idx, (l1, l2) in enumerate(zip(source_sentences, target_sentences)):
+            result = {"translation": {source: l1, target: l2}}
+            # Make sure that both translations are non-empty.
+            if all(result.values()):
+                yield idx, result
```

Comment:
```suggestion
            yield idx, result
```
the if statement is always true here

If you want to check for translations to be empty you should to 
```python
if len(l1) > 0 and len(l2) > 0:
    # do something
```